h5netcdf/core.py

```diff
@@ -17,13 +17,20 @@ def _reverse_dict(dict_):
 
 class BaseVariable(object):
 
-    def __init__(self, parent, h5ds, dimensions=None):
+    def __init__(self, parent, name, dimensions=None):
         self._parent = parent
-        self._h5ds = h5ds
+        self._root = parent._root
+        self._h5path = '/'.join([parent.name.rstrip('/'), name.lstrip('/')])
```

Comment:
I was wondering particularly about the `rstrip`/`lstrip` part. This seems fine, but let's move this logic (which is repeated twice) to a little utility function, e.g., `_join_h5paths`.